snapcraft/services/lifecycle.py

```diff
@@ -36,5 +68,47 @@ def setup(self) -> None:
             base=project.get_effective_base(),
             extra_build_snaps=project.get_extra_build_snaps(),
         )
-
         super().setup()
+
+    @overrides
+    def run(self, step_name: str | None, part_names: list[str] | None = None) -> None:
+        super().run(step_name, part_names)
+
+        enable_manifest = utils.strtobool(os.getenv("SNAPCRAFT_BUILD_INFO", "n"))
+        if enable_manifest:
+            self._generate_manifest()
+
+    def _generate_manifest(self) -> None:
```

Comment:
we should definitely have tests for the un-covered lines in this PR

Reply:
Testing for more fields. Override scripts can be added after the multi-line representation is fixed.

Reply:
At this moment yes. It is checked by review tools and a selection of fields are also expected to match.